packaging: add opensuse and sles

illescad opened the pull request:
Support for opensuse-leap 15.6 and sles 15.7 pacakging.  See [fluent-bit/10978](https://github.com/fluent/fluent-bit/pull/10978) for implementation, resolves [#10875](https://github.com/fluent/fluent-bit/issues/10875)



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Documentation**
  * Added comprehensive installation guide for openSUSE Leap and SUSE Linux Enterprise Server, including repository configuration, installation steps, and troubleshooting for common issues.
  * Updated installation documentation index with openSUSE/SLES entry.

<!-- review_stack_entry_start -->

[![Review Change Stack](https://storage.googleapis.com/coderabbit_public_assets/review-stack-in-coderabbit-ui.svg)](https://app.coderabbit.ai/change-stack/fluent/fluent-bit-docs/pull/2103?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

<!-- review_stack_entry_end -->

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

illescad committed:
packaging: add opensuse and sles

Signed-off-by: Dan Illescas <daniel.illescas@hpe.com>

esmerel commented on installation/downloads/linux/suse.md:
```diff
@@ -0,0 +1,140 @@
+
+## SLES
+
+For openSUSE and SUSE Linux Enterprise Server (SLES)
```
```suggestion
For openSUSE and SUSE Linux Enterprise Server (SLES):
```

esmerel commented on installation/downloads/linux/suse.md:
```diff
@@ -0,0 +1,140 @@
+
+For openSUSE and SUSE Linux Enterprise Server (SLES)
+
+Fluent Bit provides packages for openSUSE (Leap) and SUSE Linux Enterprise Server (SLES). The repository uses the $releasever variable to dynamically fetch packages for your specific system version.
```
```suggestion
Fluent Bit provides packages for openSUSE (Leap) and SUSE Linux Enterprise Server (SLES). The repository uses the `$releasever` variable to dynamically fetch packages for your specific system version.
```

eschabell replied:
outdated, resolving.

esmerel reviewed (approved):
I made some consistency suggestions which will fix some of the vale warnings.

I also updated the vale test so that SUSE/SLES shouldn't give warnings anymore, but those are in master so this branch is still going to have the warnings but won't after merging (or if it's rebased).

illescad committed:
packaging: fix suse.md

Signed-off-by: Dan Illescas <daniel.illescas@hpe.com>

cosmo0920 reviewed (commented):
I added a very small nitpick. Could you check it?

illescad commented on installation/downloads/linux/suse.md:
```diff
@@ -0,0 +1,137 @@
+
+- `x86_64`
+- `aarch64`
+- `arm64v8`
```
I agree, using aarch64 would be better going forward and is the standard. I can push an update. Other linux distro documentation here uses arm64v8 which may be confusing.

Also noting that the packaging scripts heavily use arm64v8 as a naming convention.

illescad committed:
packaging: only use aarch64 as supported suse architecture to avoid confusion and issues going forward

Signed-off-by: Dan Illescas <daniel.illescas@hpe.com>

cosmo0920 reviewed (approved):
For technical terminology, it looks good to me.

esmerel commented on installation/downloads/linux/suse.md:
```diff
@@ -0,0 +1,136 @@
+
+## Ensure you select the correct openSUSE verse SLES Package
+
+The openSUSE package is built and tested specifically for openSUSE Leap environments, ensuring compatibility with openSUSE libraries, update cycles, and system dependencies. Using the openSUSE package on openSUSE Leap systems helps avoid potential issues with mismatched dependencies or unsupported features that may arise from using SLES packages.
```
```suggestion
The openSUSE package is built and tested specifically for openSUSE Leap environments, ensuring compatibility with openSUSE libraries, update cycles, and system dependencies. Using the openSUSE package on openSUSE Leap systems helps avoid potential issues with mismatched dependencies or unsupported features that might arise from using SLES packages.
```

esmerel commented on installation/downloads/linux/suse.md:
```diff
@@ -0,0 +1,136 @@
+
+### Zypper install fails with incompatible dependencies
+
+OpenSUSE may track more recent dependencies than SLES. If you encounter dependency issues, ensure you are using the correct repository path for your OS distro.
```
```suggestion
OpenSUSE might track more recent dependencies than SLES. If you encounter dependency issues, ensure you are using the correct repository path for your OS distro.
```

esmerel committed:
Apply suggestions from code review

Signed-off-by: Lynette  Miles <6818907+esmerel@users.noreply.github.com>

eschabell commented on installation/downloads/linux/suse.md:
````diff
@@ -0,0 +1,140 @@
+```shell
+$ sed -i 's/mirrorlist/#mirrorlist/g' /etc/yum.repos.d/CentOS-* && \
+
+$ sed -i 's|#baseurl=http://mirror.centos.org|baseurl=http://vault.centos.org|g' /etc/yum.repos.d/CentOS-*
````
outdated, resolving.

eschabell committed:
Merge branch 'master' into illescad-packaging-suse

Resolve conflicts:
- SUMMARY.md: insert openSUSE/SLES into master's alphabetized Linux list
- installation/downloads.md: master replaced the table; openSUSE/SLES
  added to the new Linux bullet list instead
- installation/getting-started-with-fluent-bit.md: accept upstream
  deletion (content consolidated into downloads.md)

Also fix one vale error in suse.md (may -> might).

Signed-off-by: Eric D. Schabell <eric@schabell.org>

eschabell committed:
packaging: suse: address vale findings

CI vale runs with -level=info -fail-on-error=true and flagged the SUSE
docs. Fix without changing vale-styles:

- Backtick command and variable names (`zypper`, `$releasever`,
  `baseproduct`) so the Spelling rule treats them as code spans.
- Backtick brand names (`openSUSE`, `SLES`, `SUSE Linux Enterprise
  Server`) in headings so the Headings capitalization rule treats them
  as masked tokens.
- Fix typo in heading ("verse" -> "or"); lowercase trailing "Package".
- "is not recommended" -> "isn't recommended" (contraction).
- Expand casual abbreviations ("repo" -> "repository",
  "distro" -> "distribution").

Signed-off-by: Eric D. Schabell <eric@schabell.org>

eschabell reviewed (approved):
@illescad now just waiting on code PR merging.
